File: IdadedeCachorro/app/src/main/java/annia/com/br/idadedecachorro/MainActivity.java
package annia.com.br.idadedecachorro;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    private EditText caixaTexto;
    private Button botaoIdade;
    private TextView resultadoIdade;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        caixaTexto = findViewById(R.id.activity_text_idade);
        botaoIdade = findViewById(R.id.activity_button_calcular);
        resultadoIdade = findViewById(R.id.activity_text_resultado);

        botaoIdade.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                //Recuperar o que foi digitado
                String textoDigitado = caixaTexto.getText().toString();

                if(textoDigitado.isEmpty()) {
                    //String vazia (mensagem de erro)
                    Toast.makeText(MainActivity.this, "Por favor, digite uma idade para calcular", Toast.LENGTH_SHORT).show();
                } else {
                    int valorDigitado = Integer.parseInt(textoDigitado);
                    int resultadoFinal = valorDigitado*7;

                    resultadoIdade.setText("A idade do cachorro em anos humanos é: " + resultadoFinal);
                }

            }
        });

    }
}

File: IdadedeCachorroKotlin/app/src/main/java/com/example/idadedecachorrokotlin/MainActivity.kt
package com.example.idadedecachorrokotlin

import android.content.Context
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    private val TITULO_APPBAR:String = "Idade de Cachorro Kotlin"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        setTitle(TITULO_APPBAR)

        val caixaTexto:EditText = activity_editText_idade
        val calcularIdade:Button = activity_button_calcularIdade
        val resultado:TextView = activity_text_idade

        calcularIdade.setOnClickListener() {
            val textoDigitado:String? = caixaTexto.text.toString()

            val valorDigitado = if (textoDigitado.isNullOrEmpty())
                                    Toast.makeText(this,"Digite um texto", Toast.LENGTH_LONG).show()
                                else
                                    (resultado.setText("Idade: " + textoDigitado.toInt()*7))
        }

    }
}
